test/src/error-snippets/frames.ts

```diff
@@ -0,0 +1,141 @@
+                "../utils/logger": logger,
+            }).findRelevantStackFrame;
+        });
+
+        it("should return null without throwing exceptions on bad input", () => {
+            const stackFrame = findRelevantStackFrame("foo" as unknown as Error);
+
+            assert.calledOnceWith(logger.warn, "Unable to find relevant stack frame:");
+            assert.isNull(stackFrame);
+        });
+
+        it("should not return repl frames", () => {
+            const error = buildError_(frames.browser.repl, frames.node.repl);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            console.log(stackFrame);
+            assert.notCalled(logger.warn);
+            assert.isNull(stackFrame);
+        });
+
+        it("should not return internal frames", () => {
+            const error = buildError_(frames.node.internal1, frames.node.internal2);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.notCalled(logger.warn);
+            assert.isNull(stackFrame);
+        });
+
+        it("should return wdio frame, if there is nothing left", () => {
+            const error = buildError_(frames.node.internal1, frames.node.repl, frames.browser.wdio);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.equal(stackFrame?.functionName, "async Element.elementErrorHandlerCallbackFn");
+        });
+
+        it("should return first seen frame with equal relevance", () => {
+            const error = buildError_(frames.node.internal2, frames.node.repl, frames.node.wdio, frames.browser.wdio);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.equal(stackFrame?.functionName, "Element.elementErrorHandlerCallbackFn");
+        });
+
+        it("should value other project deps frame more, than wdio frame", () => {
+            const error = buildError_(frames.browser.wdio, frames.browser.unknownWdio, frames.node.projectDeps);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.equal(stackFrame?.functionName, "Object.tryCatcher");
+        });
+
+        it("should value user frame above all for browser", () => {
+            const error = buildError_(
+                frames.browser.wdio,
+                frames.browser.projectDeps,
+                frames.browser.repl,
+                frames.browser.user,
+            );
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.equal(stackFrame?.functionName, "async Object.<anonymous>");
+        });
+
+        it("should value user frame above all for node", () => {
+            const error = buildError_(frames.node.wdio, frames.node.unknownWdio, frames.node.repl, frames.node.user);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.equal(stackFrame?.functionName, "Object.<anonymous>");
+        });
+    });
+
+    it("resolveWithStackFrame", () => {
```

Comment:
not the most understandable test name :)

Reply:
`stackFrameResolver` -> `stackFrameLocationResolver`
`resolveWithSourceMap` -> `resolveLocationWithSourceMap`
`resolveWithStackFrame` -> `resolveLocationWithStackFrame`